Neos.Flow/Classes/Log/ThrowableStorage/FileStorage.php

```diff
@@ -233,4 +255,35 @@ protected function renderRequestInfo()
 
         return $output;
     }
+
+    /**
+     * Cleans up existing throwable dumps when they are older than the configured
+     * maximum age or the oldest ones exceeding the maximum number of dumps allowed.
+     */
+    protected function cleanupThrowableDumps(): void
+    {
+        if ($this->maximumThrowableDumpAge > 0) {
+            $cutoffTime = time() - $this->maximumThrowableDumpAge;
+
+            /** @var \SplFileInfo $directoryEntry */
+            $iterator = new \DirectoryIterator($this->storagePath);
+            foreach ($iterator as $directoryEntry) {
+                if ($directoryEntry->isFile() && $directoryEntry->getCTime() < $cutoffTime) {
+                    unlink($directoryEntry->getRealPath());
+                }
+            }
+        }
+
+        if ($this->maximumThrowableDumpCount > 0) {
+            // this returns alphabetically ordered, so newest first, as we have a date/time in the name
+            $existingDumps = glob($this->storagePath);
+            if (count($existingDumps) > $this->maximumThrowableDumpCount) {
+                $dumpsToRemove = array_slice($existingDumps, $this->maximumThrowableDumpCount);
```

Comment:
Yes, I don't care at that point. Because: The dumps files

- are ordered by name, see , where it says
   > GLOB_NOSORT -  […]. When this flag is not used, the pathnames are sorted alphabetically 
- and the name starts with a timestamp generated through `date('YmdHis', $time)`
- so the older files come first

Which makes the code almost correct… 😇 The maximum should be left, not removed, so it should be

```php
$dumpsToRemove = array_slice($existingDumps, -$this->maximumThrowableDumpCount);
```

to slice from the end of the array, see https://3v4l.org/FUGWC

Reply:
And now of course those are the ones that should **not** be deleted. 🤪

Reply:
Like this: https://3v4l.org/46LTA